src/utils/internal.ts

```diff
@@ -1174,3 +1197,237 @@ export function assertTickerValid(ticker: string): void {
+    return currentOwnership.eq(value as BigNumber);
+  } else if (transferRestriction.isClaimCount && type === TransferRestrictionType.ClaimCount) {
+    const [statClaim, rawIssuerId, rawMin, maybeMax] = transferRestriction.asClaimCount;
+    const issuerDid = identityIdToString(rawIssuerId);
+    const min = u64ToBigNumber(rawMin);
+    const max = maybeMax.isSome ? u64ToBigNumber(maybeMax.unwrap()) : undefined;
+    const castedValue = value as ClaimCountRestrictionValue;
+
+    return !!(
+      castedValue.min.eq(min) &&
+      compareOptionalBigNumbers(max, castedValue.max) &&
+      castedValue.claim.type === getClaimType(statClaim) &&
+      issuerDid === castedValue.issuer.did
+    );
+  } else if (
+    transferRestriction.isClaimOwnership &&
+    type === TransferRestrictionType.ClaimPercentage
+  ) {
+    const castedValue = value as ClaimCountRestrictionValue;
+    const [statClaim, rawIssuerId, rawMin, rawMax] = transferRestriction.asClaimOwnership;
+    const issuerDid = identityIdToString(rawIssuerId);
+
+    const min = permillToBigNumber(rawMin);
+    const max = permillToBigNumber(rawMax);
+
+    return !!(
+      castedValue.min.eq(min) &&
+      compareOptionalBigNumbers(castedValue.max, max) &&
+      castedValue.claim.type === getClaimType(statClaim) &&
+      issuerDid === castedValue.issuer.did
+    );
+  }
+
+  return false;
+}
+
+/**
+ * @hidden
+ */
+export function compareStatTypeToTransferRestrictionType(
+  statType: PolymeshPrimitivesStatisticsStatType,
+  transferRestrictionType: TransferRestrictionType
+): boolean {
+  const opType = meshStatToStatisticsOpType(statType);
+  if (opType === StatisticsOpType.Count) {
+    return transferRestrictionType === TransferRestrictionType.Count;
+  } else if (opType === StatisticsOpType.Balance) {
+    return transferRestrictionType === TransferRestrictionType.Percentage;
+  } else if (opType === StatisticsOpType.ClaimCount) {
+    return transferRestrictionType === TransferRestrictionType.ClaimCount;
+  } else {
+    return transferRestrictionType === TransferRestrictionType.ClaimPercentage;
+  }
+}
+
+/**
+ *@ hidden
+ * @param args.type TransferRestriction type that was given
+ * @param args.claimIssuer optional Issuer and ClaimType for the scope of the Stat
+ * @param context
+ * @returns encoded StatType needed for the TransferRestriction to be enabled
+ */
+export function neededStatTypeForRestrictionInput(
+  args: { type: TransferRestrictionType; claimIssuer?: ClaimIssuer },
+  context: Context
+): PolymeshPrimitivesStatisticsStatType {
+  const { type, claimIssuer } = args;
+
+  let rawOp;
+  if (type === TransferRestrictionType.Count || type === TransferRestrictionType.ClaimCount) {
+    rawOp = statisticsOpTypeToStatOpType(StatisticsOpType.Count, context);
+  } else {
+    rawOp = statisticsOpTypeToStatOpType(StatisticsOpType.Balance, context);
+  }
+
+  const rawIssuer = claimIssuer ? claimIssuerToMeshClaimIssuer(claimIssuer, context) : undefined;
```

Comment:
Can't you use optionize here?

Reply:
(I'm assuming `claimIssuer` is an `Option`)

Reply:
It is, but its actually Option<ITuple>, which made it trickier than dealing with just a normal tuple and letting polkadot.js deal with it.

We could look into making an ITuple conversion util (or maybe I am missing it), but for now I think this is easier.

for reference raw type is: 
```
Option<
      ITuple<[PolymeshPrimitivesIdentityClaimClaimType, PolymeshPrimitivesIdentityId]>
 >;
```
and I get `[PolymeshPrimitivesIdentityClaimClaimType, PolymeshPrimitivesIdentityId]` doesn't implement Codec error